### PR TITLE
[#257] Graceful degradation when API unavailable

### LINKED ISSUE #257: [UX] Graceful degradation when API unavailable
## Problem
If Claude API is down, rate-limited, or key is invalid, users get Python tracebacks and Cortex is unusable.

## Solution
Graceful fallback chain:

```
$ cortex install docker
⚠️  AI unavailable (rate limited). Falling back to smart search...

Found packages matching 'docker':
  1. docker.io - Container runtime
  2. docker-compose - Multi-container orchestration
  3. docker-buildx - Extended build capabilities

Select [1-3] or 'q' to quit:
```

## Fallback Chain
1. Try Claude API
2. If fail → Try cached responses (if any)
3. If fail → apt-cache search with smart ranking
4. If fail → Show clear error with fix instructions

## Acceptance Criteria
- [ ] Catch all API exceptions (rate limit, auth, network, timeout)
- [ ] User-friendly error messages (no tracebacks)
- [ ] Fallback to apt-cache search
- [ ] Cache successful API responses for offline use
- [ ] Clear messaging about degraded mode

## Bounty: $50

### PR DESCRIPTION
**Pull request by mikejmorgan-ai**
Implements #257 with complete tests and docs. Closes #257

## Summary by CodeRabbit

* **New Features**
  * Added graceful degradation system enabling offline functionality with multi-tier fallbacks: LLM response caching, pattern-based command generation, and manual guidance when APIs are unavailable.

* **Documentation**
  * Added comprehensive guide for graceful degradation with usage examples, API reference, configuration options, and troubleshooting.

* **Tests**
  * Added extensive test suite for graceful degradation module covering all functionality and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: tests/test_graceful_degradation.py [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+    
+    def test_check_api_health_no_key(self, manager):
+        """Test health check when no API key is set."""
+        with patch.dict(os.environ, {}, clear=True):
+            # Remove keys if they exist
+            os.environ.pop("ANTHROPIC_API_KEY", None)
+            os.environ.pop("OPENAI_API_KEY", None)
+            
+            result = manager.check_api_health()
+            assert result.status == APIStatus.UNAVAILABLE
```
[the remark this answers is not in the document]
Comment (shanko) [reply tests/test_graceful_degradation.py]: I cannot ascertain the validity of this recommendation. Use another AI tool (or a human expert) to get a second opinion.

File: cortex/graceful_degradation.py [hunk 2]
```diff
@@ -0,0 +1,538 @@
...
+    """SQLite-based cache for LLM responses."""
+    
+    def __init__(self, db_path: Optional[Path] = None):
+        self.db_path = db_path or Path.home() / ".cortex" / "response_cache.db"
+        self.db_path.parent.mkdir(parents=True, exist_ok=True)
+        self._init_db()
+    
+    def _init_db(self):
+        """Initialize the cache database."""
+        with sqlite3.connect(self.db_path) as conn:
```
Comment (shanko) [inline cortex/graceful_degradation.py]: Consider caching the connection object in the constructor of this class. That will avoid repeated sqlite3.connect() calls in the multiple methods of the class.

File: tests/test_graceful_degradation.py [hunk 3]
```diff
@@ -0,0 +1,453 @@
+"""
+Tests for Graceful Degradation Module
+
+Issue: #257
+"""
+
+import pytest
+import tempfile
```
Comment (shanko) [inline tests/test_graceful_degradation.py]: All recommendation of removing unused imports are valid and should be implemented.
